polars/polars-core/src/fmt.rs

```diff
@@ -338,6 +338,7 @@ impl Display for DataFrame {
             .unwrap_or_else(|_| "8".to_string())
             .parse()
             .unwrap_or(8);
+        let max_n_rows = if max_n_rows < 2 { 2 } else { max_n_rows };
```

Comment:
See the conditional compilation in line 336. If we compile without default features  `max_n_rows` doesn't exist.